pylint/extensions/dict_init_mutate.py

```diff
@@ -0,0 +1,61 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Check for use of dictionary mutation after initialization."""
+
+from astroid import nodes
+
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import only_required_for_messages
+from pylint.interfaces import HIGH
+from pylint.lint.pylinter import PyLinter
+
+
+class DictInitMutateChecker(BaseChecker):
+    name = "dict-init-mutate"
+    msgs = {
+        "W3301": (
+            "Dictionary mutated immediately after initialization",
```

Comment:
I think this message could be more explicit about the solution and the problem, but I'm not sure. We added a function to print big suggestions of dict recently, so maybe it's possible to do something like
```suggestion
            "Declare all known values directly in the constructor with '%s'",
```
Ie. result would be ``Declare all known values directly in the constructor with ' {"apple": 1, "banana": 10}'``  ``Declare all known values directly in the constructor with ' {"apple": 1, "melon" : 4, ... "banana": 10}'``

Reply:
I like the updated language, but the suggestion with %s may be tricky because it may be multiple lines after. I'll see what I can do...